File: BE-FindingJob/src/main/java/com/example/befindingjob/service/JobService.java
package com.example.befindingjob.service;

import com.example.befindingjob.dto.ItemJobDTO;
import com.example.befindingjob.dto.JobDTO;
import com.example.befindingjob.entity.Job;
import com.example.befindingjob.model.ApiResponse;
import org.checkerframework.checker.units.qual.A;

import java.util.List;

public interface JobService {
    ApiResponse<Void> createJob(JobDTO jobDTO);
    ApiResponse<Void> updateJob(Job job);
    ApiResponse<Void> deleteJob(int jobId);
    ApiResponse<List<ItemJobDTO>> getAllJobsByUser(String token);
}

File: JobHub/app/src/main/java/com/example/jobhub/service/JobService.kt
package com.example.jobhub.service

import com.example.jobhub.dto.ItemJobDTO
import com.example.jobhub.dto.JobDTO
import com.example.jobhub.entity.Job
import com.example.jobhub.model.ApiResponse
import retrofit2.Call
import retrofit2.http.Body
import retrofit2.http.GET
import retrofit2.http.Header
import retrofit2.http.POST
import retrofit2.http.Path
import retrofit2.http.Query

interface JobService {
    @POST("api/job/create-job")
    fun createJob(@Body jobDTO: JobDTO): Call<ApiResponse<Void>>

    @POST("api/job/update-job")
    fun updateJob(@Body job: Job): Call<ApiResponse<Void>>

    @POST("api/job/delete-job")
    fun deleteJob(@Body jobId: Int): Call<ApiResponse<Void>>

    @POST("api/job/get-all-jobs-by-user")
    fun getAllJobsByUser(@Header("token") token: String): Call<ApiResponse<List<ItemJobDTO>>>

    @GET("api/job/{id}")
    fun getJobById(
        @Header("token") token: String,
        @Path("id") id: Int
    ): Call<ApiResponse<JobDTO>>
}